subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/StartParameterResolutionOverride.java

```diff
@@ -243,7 +243,7 @@ private FailureVerificationOverride(Exception error) {
 
         @Override
         public ModuleComponentRepository overrideDependencyVerification(ModuleComponentRepository original, String resolveContextName, ResolutionStrategyInternal resolutionStrategy) {
-            throw new GradleException("Dependency verification cannot be performed", error);
+            throw new TaskDependencyResolveException("Dependency verification cannot be performed", error);
```

Comment:
Why use that type here? It feels somewhat unrelated.

Reply:
This is not a mandatory change. I just wanted to throw a `Contextual` exception here, but it doesn't actually change anything (besides some arguable consistency), so if that type is not ok, I can just revert this.